lib/rules/declaration-block-no-redundant-longhand-properties/index.js

```diff
@@ -22,8 +22,49 @@ const meta = {
 	fixable: true,
 };
 
+const customResolvers = new Map([
+	[
+		'grid-template',
+		(/** @type {Map<string, Declaration>} */ transformedDeclarationNodes) => {
+			const areas = transformedDeclarationNodes.get('grid-template-areas')?.value.trim();
+			const columns = transformedDeclarationNodes.get('grid-template-columns')?.value.trim();
+			const rows = transformedDeclarationNodes.get('grid-template-rows')?.value.trim();
+
+			if (areas === undefined || columns === undefined || rows === undefined) return;
```

Comment:
[suggest] Excluding also `''` seems safer because of using `.trim()` like this:

```suggestion
			if (!(areas && columns && rows)) return;
```